es-space-mgmt/main/default/lwc/reservationTile/__tests__/reservationTile.test.js

```diff
@@ -0,0 +1,123 @@
+            );
+            expect(paragraphElements[2].textContent).toBe(
+                'Market: ' + RESERVATION_LEAD.record.Market__r.Name
+            );
+            expect(paragraphElements[5].textContent).toBe(
+                'Total Guests: ' +
+                    RESERVATION_LEAD.record.Total_Number_of_Guests__c
+            );
+            // check value of formatted date elements
+            const datetimeElements = element.shadowRoot.querySelectorAll(
+                'lightning-formatted-date-time'
+            );
+            expect(datetimeElements[0].value).toBe(
+                RESERVATION_LEAD.record.Start_Date__c
+            );
+            expect(datetimeElements[1].value).toBe(
+                RESERVATION_LEAD.record.End_Date__c
+            );
+        });
+    });
+
+    it('render reservationTile for Contact', () => {
+        const element = createElement('c-reservation-tile', {
+            is: ReservationTile
+        });
+        element.reservation = RESERVATION_CONTACT;
+        document.body.appendChild(element);
+        return Promise.resolve().then(() => {
+            // check for div class
+            const divElement = element.shadowRoot.querySelector('div.pointer');
+            expect(divElement.className).toBe('pointer');
+            // check for paragraph text nodes
+            const paragraphElements = element.shadowRoot.querySelectorAll('p');
+            expect(paragraphElements[0].textContent).toBe(
+                'Customer: ' + RESERVATION_CONTACT.record.Contact__r.Name
+            );
+            expect(paragraphElements[1].textContent).toBe(
+                'Status: ' + RESERVATION_CONTACT.record.Status__c
+            );
+            expect(paragraphElements[2].textContent).toBe(
+                'Market: ' + RESERVATION_CONTACT.record.Market__r.Name
+            );
+            expect(paragraphElements[5].textContent).toBe(
+                'Total Guests: ' +
+                    RESERVATION_CONTACT.record.Total_Number_of_Guests__c
+            );
+            // check value of formatted date elements
+            const datetimeElements = element.shadowRoot.querySelectorAll(
+                'lightning-formatted-date-time'
+            );
+            expect(datetimeElements[0].value).toBe(
+                RESERVATION_CONTACT.record.Start_Date__c
+            );
+            expect(datetimeElements[1].value).toBe(
+                RESERVATION_CONTACT.record.End_Date__c
+            );
+        });
+    });
+
+    it('click on reservation tile', () => {
+        const element = createElement('c-reservation-tile', {
+            is: ReservationTile
+        });
+        element.reservation = RESERVATION_CONTACT;
+        document.body.appendChild(element);
+        // listen to flow navigate event
+        const handler = jest.fn();
+        element.addEventListener('reservationselect', handler);
+        return Promise.resolve()
```

Comment:
No need to put stuff into a Promise here. You would need the Promise only for checking thing after a UI update. As everything is rendered in this component directly when you append the element to the DOM (as in: you don't have any wire or value setting in your component) its not needed.

Reply:
fixed!